api/handlers.go

```diff
@@ -71,7 +77,61 @@ func (s *Server) Pulse(ctx echo.Context, pulseNumber server.PulseNumberPathParam
 }
 
 func (s *Server) JetDropsByPulseNumber(ctx echo.Context, pulseNumber server.PulseNumberPathParam, params server.JetDropsByPulseNumberParams) error {
-	panic("implement me")
+	var failures []server.CodeValidationFailures
+	if !pulse.IsValidAsPulseNumber(int(pulseNumber)) {
+		failures = append(failures, server.CodeValidationFailures{
+			FailureReason: NullableString("invalid"),
+			Property:      NullableString("pulse"),
+		})
+	}
+	limit, offset, err := checkLimitOffset(params.Limit, params.Offset)
+	if err != nil {
+		failures = append(failures, server.CodeValidationFailures{
+			FailureReason: NullableString("invalid"),
+			Property:      NullableString("limit or offset"),
+		})
+	}
+
+	jetDropID, err := checkJetDropID(params.FromJetDropId)
+	if err != nil {
+		failures = append(failures, server.CodeValidationFailures{
+			FailureReason: NullableString("invalid"),
+			Property:      NullableString("jet drop id"),
+		})
+	}
+
+	if failures != nil {
+		response := server.CodeValidationError{
+			Code:               NullableString(strconv.Itoa(http.StatusBadRequest)),
+			Description:        nil,
+			Link:               nil,
+			Message:            NullableString(InvalidParamsMessage),
+			ValidationFailures: &failures,
+		}
+		return ctx.JSON(http.StatusBadRequest, response)
+	}
+
+	jetDrops, total, err := s.storage.GetJetDropsWithParams(
+		models.Pulse{PulseNumber: int(pulseNumber)},
+		jetDropID,
+		limit,
+		offset,
+	)
+	if err != nil {
+		s.logger.Error(err)
```

Comment:
Don't we have to return err message in this case to user?

Reply:
I thought that we hide internal errors from user to log, am I wrong ?